theseus/geometry/functional/lie_group.py

```diff
@@ -26,6 +26,18 @@ def _jinverse_impl(group: torch.Tensor) -> Tuple[List[torch.Tensor], torch.Tenso
     return _jinverse_impl
 
 
+def LeftProjectImplFactory(module):
+    def _left_project_impl(group: torch.Tensor, matrix: torch.Tensor) -> torch.Tensor:
+        if not module.check_group_tensor(group):
+            raise ValueError("Invalid data tensor for SO3.")
```

Comment:
In the Lie group file should this error message be specific to SO3? This might also be related to a comment in another PR about moving the error message inside the `check_group_tensor` function.